chainerx_cc/chainerx/routines/math_test.cc

```diff
@@ -1645,6 +1645,35 @@ TEST_P(MathTest, LogSoftmaxDoubleBackward) {
             [](const std::vector<Array>& xs) -> std::vector<Array> { return {LogSoftmax(xs[0])}; }, {a}, {go}, {ggi}, {eps, eps});
 }
 
+TEST_THREAD_SAFE_P(MathTest, Square) {
+    Array a = testing::BuildArray({3, 1}).WithData<float>({-1.f, 2.f, 0.f});
+    Array e = testing::BuildArray({3, 1}).WithData<float>({(-1.f) * (-1.f), (2.f) * (2.f), (0.f) * (0.f)});
+
+    Run([&]() { testing::CheckForward([](const std::vector<Array>& xs) { return std::vector<Array>{Square(xs[0])}; }, {a}, {e}); });
+}
+
+TEST_P(MathTest, SquareBackward) {
+    using T = double;
+    Shape shape{2, 3};
+    Array a = (*testing::BuildArray(shape).WithLinearData<T>(1).WithPadding(1)).RequireGrad();
+    Array go = testing::BuildArray(shape).WithLinearData<T>(-0.1, 0.1).WithPadding(1);
+    Array eps = Full(shape, 1e-3);
```

Comment:
This test seems to be failing due to wrong eps dtype.

You actually need to specify it explicitly to be a `Dtype::kFloat64` similar to here https://github.com/chainer/chainer/pull/6486/files#diff-ccbc4c392cace589c74e9304ecc6b99aR1642 because it will otherwise default to the default dtype which is `Dtype::kFloat32`.

Note that you can take a look at the CI outputs https://jenkins.preferred.jp/job/chainer/job/chainer_pr/63/

Reply:
Thanks @hvy . Made the fixes.